src/Editor.js

```diff
@@ -307,11 +308,22 @@ const Editor = () => {
       );
       break;
     case "drag":
+      const gradient =
+        mouseOnShiftDown &&
+        (mouse.y - mouseOnShiftDown.y) / (mouse.x - mouseOnShiftDown.x);
```

Comment:
We shouldn't need to track `mouseOnShiftDown` specifically, we care about where the user's mouse was when they started dragging, not where it was when they pressed shift:

```js
and(keys, ["ShiftLeft", "ShiftRight"]).size > 0 &&
(mouse.y - mouse.startY) / (mouse.x - mouse.startX)
```

However, I think that's going to cause another problem because `keys` isn't declared until further down. I'm suspecting that `useKeys` hook needs a big ol' rewrite… but it probably needs some more thought before making any rash decisions. 😄

Reply:
I understand the not needing to track ```mouseOnShiftDown```  but I don't get why ```keys``` needs to be declared further down (where it is right now). 

Why do you think the ```useKeys``` hook needs a rewrite? I don't understand this

Reply:
Oh! If it's possible to move it up without any issues then that's perfect. I just assumed it was relying on `transformedLayers` and we'd run into a chicken-and-egg situation. Happy days if that's not a problem. 😄